### PR TITLE
feat(ProtoViewBuilder): throw for invalid event names

### PR DESCRIPTION
**Pull request by cironunes**
This integrates #3769 to the `ProtoViewBuilder`

/cc @pkozlowski-opensource

### CODE REVIEW

File: modules/angular2/src/render/dom/view/proto_view_builder.ts [hunk 1]
```diff
@@ -368,10 +369,13 @@ function buildElementPropertyBindings(
   return propertyBindings;
 }
 
-function isValidElementPropertyBinding(schemaRegistry: ElementSchemaRegistry,
+function isValidElementPropertyBinding(schemaRegistry: DomElementSchemaRegistry,
```
Comment (pkozlowski-opensource) [inline modules/angular2/src/render/dom/view/proto_view_builder.ts]: This sounds wrong and shouldn't be necessery

File: modules/angular2/src/render/dom/view/proto_view_builder.ts [hunk 2]
```diff
@@ -368,10 +369,13 @@ function buildElementPropertyBindings(
...
                                        protoElement: /*element*/ any, isNgComponent: boolean,
                                        binding: ElementPropertyBinding): boolean {
   if (binding.type === PropertyBindingType.PROPERTY) {
+    if (!schemaRegistry.hasEvent(protoElement, binding.property)) {
+      throw new BaseException(`Can't bind to '${binding.property}' since it isn' a valid custom event name`);
```
Comment (pkozlowski-opensource) [inline modules/angular2/src/render/dom/view/proto_view_builder.ts]: I'm confused... This part of the code is about binding properties, why the the crux of the issue was to check for events...

**Comment by pkozlowski-opensource** [comment]
@cironunes Thnx for the PR, but I think that there is some misunderstanding about what we want to achieve here - we want to validate event names, not property names. I've also left some other comments in the code.

We can't merge it in the current form and different change is needed here, so I'm going to close this PR for now. Please open a new one when you've got it sorted out.

File: modules/angular2/src/render/dom/view/proto_view_builder.ts (hunk 1 above)
Comment (cironunes) [inline modules/angular2/src/render/dom/view/proto_view_builder.ts]: Indeed.

**Pull request closed without merging**